### PR TITLE
Improve Makefile mechanics & output

### PR DESCRIPTION
**Pull request by fornellas**
Improve Makefile:

- Nicer slimmer output.
- Travis CI signal remains full verbose.
- Colored output works on fails.
- Fixed mypy not covering `tests/`.
- Moved `mock_async_callable` tests to its own file (as is).
- Making a test file now works: `make tests/cli_unittest.py`.

And finally, `make -j`:

- Before: 19.6s.
- Now: 9.0s.

# Quiet on success

image

# Show output on failures

image

# Show output with `V=1`

image

### CODE REVIEW

File: tests/mock_async_callable_testslide.py [hunk 1]
```diff
@@ -0,0 +1,638 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
```
Comment (david-caro) [inline tests/mock_async_callable_testslide.py:1]: Can you split this on it's own PR? It looks unrelated (maybe I'm missing something xd)

File: util/run_silent_if_successful.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+os.close(slave_pty_fd)
+
+master_pty_fd_reader_thread.join()
+
+if completed_process.returncode:
+    for data in read_data:
+        os.write(sys.stdout.fileno(), data)
+    sys.exit(completed_process.returncode)
```
Comment (david-caro) [inline util/run_silent_if_successful.py:38]: We might want to save the logs in a file or something eventually, for debugging after the fact.

File: tests/mock_async_callable_testslide.py (hunk 1 above)
Comment (fornellas) [reply tests/mock_async_callable_testslide.py]: o7

File: util/run_silent_if_successful.py (hunk 2 above)
Comment (fornellas) [reply util/run_silent_if_successful.py:38]: One can do `make V=1` to see all output, but I also see values on having logs.

I'm not sure how we'd implement it: `Makefile` has many different rules, and we'd need a way to reset the log at every execution. Plus, there's concurrency (`make -j`) to consider.
